utils/fragment_range.hh

```diff
@@ -410,3 +410,23 @@ void write_native(Out& out, std::type_identity_t<T> v) {
         write_fragmented(out, single_fragmented_view(bytes_view(p, sizeof(v))));
     }
 }
+
+static inline sstring::iterator fragment_to_hex(sstring::iterator out, bytes_view frag) {
```

Comment:
`static inline` shouldn't be used in C++ headers. It causes each translation unit to instantiate the function separately, and if the compiler chooses not to inline, we end up with one copy per translation unit that instantiates the function.

In addition, it shouldn't be in a header at all (see #1). We shouldn't have headers, but that will take more work.

Reply:
(plain `inline` allows the linker to merge duplicate instantiations)